### PR TITLE
Add support for `is`, `is!` expressions

### PR DESCRIPTION
**Pull request by thosakwe**
Should take care of #101. 😄

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,6 @@ doc/api/
 # Don't commit pubspec lock file
 # (Library packages only! Remove pattern if developing an application package)
 pubspec.lock
+
+# Avoid IntelliJ adding junk files to the repo
+.idea
```
Comment (kevmoo) [inline .gitignore]: Please, don't. These should be ignored in a user's global git ignore.

The number machine and IDE-specific files could grow forever...

Comment (thosakwe) [inline .gitignore]: Ah, my bad . Didn't realize.

**Comment by thosakwe** [comment]
@kevmoo Fixed

**Comment by kevmoo** [comment]
@matanlurey should actually help w/ content – my ovly value is making sure `.gitignore` is pretty. 🧀

File: lib/src/builders/expression.dart [hunk 2]
```diff
@@ -295,6 +296,14 @@ abstract class AbstractExpressionMixin implements ExpressionBuilder {
   }
 
   @override
+  ExpressionBuilder isInstanceOf(TypeBuilder type) =>
+      new _IsInstanceOfExpression(this, type);
+
+  @override
+  ExpressionBuilder isNotInstanceOf(TypeBuilder type) =>
```
Comment (matanlurey) [inline lib/src/builders/expression.dart]: There already is a `negate` expression, no reason to add this.

File: lib/src/builders/expression.dart [hunk 3]
```diff
@@ -447,6 +456,12 @@ abstract class ExpressionBuilder
     Map<String, ExpressionBuilder> namedArguments,
   });
 
+  /// Returns as an [ExpressionBuilder] indicating whether this expression is an instance of [type], using the `is` operator.
+  ExpressionBuilder isInstanceOf(TypeBuilder type);
+
+  /// Returns as an [ExpressionBuilder] indicating whether this expression is not an instance of [type], using the `is!` operator.
```
Comment (matanlurey) [inline lib/src/builders/expression.dart]: Ditto, remove.

File: lib/src/builders/expression/is_instance_of.dart [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  AstNode buildAst([Scope scope]) => buildExpression(scope);
+
+  @override
+  Expression buildExpression([Scope scope]) {
+    return astFactory.isExpression(
+        _expression.buildExpression(scope), $is, null, _type.buildType(scope));
+  }
+}
+
+class _IsNotInstanceOfExpression extends AbstractExpressionMixin
```
Comment (matanlurey) [inline lib/src/builders/expression/is_instance_of.dart]: Ditto remove.

File: test/builders/is_instance_of_test.dart [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import 'package:test/test.dart';
+
+final TypeBuilder _barType = new TypeBuilder('Bar');
+
+void main() {
+  test('should emit an `is` expression', () {
+    expect(reference('foo').isInstanceOf(_barType), equalsSource('foo is Bar'));
+  });
+
+  test('should emit an `is!` expression', () {
```
Comment (matanlurey) [inline test/builders/is_instance_of_test.dart]: OK to keep this test, just use negate instead.

Comment (thosakwe) [inline test/builders/is_instance_of_test.dart]: Ok, give me one minute.

Comment (thosakwe) [inline test/builders/is_instance_of_test.dart]: Sorry, got preoccupied. Fixed now!

**Comment by thosakwe** [comment]
Woohoo! 🎉

**Pull request merged**